net/miniupnpd/files/miniupnpd.init

```diff
@@ -217,6 +226,119 @@ stop_service() {
 }
 
 start_service() {
+	# Begin UCI migration of /etc/config/upnpd to v1.1
+	# Migrate stun_host and stun_port and pre-configure compatible STUN server
```

Comment:
Please put all migration stuff into a separate migration function. Call it from start_service(). 

Here you perform an in-place migration, which is acceptable. Document each moment why migrating each specific setting is necessary. The daemon no longer understands it?